extensions/stackdriver/log/exporter.cc

```diff
@@ -42,19 +42,23 @@ namespace Stackdriver {
 namespace Log {
 
 ExporterImpl::ExporterImpl(RootContext* root_context,
-                           const std::string& logging_service_endpoint)
-    : export_call_(MetricType::Counter, "stackdriver_logging_filter",
-                   {MetricTag{"type", MetricTag::TagType::String},
-                    MetricTag{"success", MetricTag::TagType::Bool}}) {
+                           const std::string& logging_service_endpoint) {
   context_ = root_context;
-  success_callback_ = [this](google::protobuf::Empty&&) {
-    export_call_.increment(1, "logging", true);
+  Metric export_call(MetricType::Counter, "stackdriver_filter",
+                     {MetricTag{"type", MetricTag::TagType::String},
+                      MetricTag{"success", MetricTag::TagType::Bool}});
+  auto success_counter = export_call.resolve("logging", true);
+  auto failure_counter = export_call.resolve("logging", false);
+  success_callback_ = [success_counter](google::protobuf::Empty&&) {
+    // TODO(bianpengyuan): replace this with envoy's generic gRPC counter.
+    incrementMetric(success_counter, 1);
     logDebug("successfully sent Stackdriver logging request");
   };
 
-  failure_callback_ = [this](GrpcStatus status, StringView message) {
+  failure_callback_ = [failure_counter](GrpcStatus status, StringView message) {
     // TODO(bianpengyuan): add retry.
-    export_call_.increment(1, "logging", false);
```

Comment:
(cc @jplevyak) I think we need a regression test for this, but I don't see a simple way to simulate this in envoy test framework. Let's file an issue to create a test case for reproducing the failure to confirm this fix works.

Reply:
What are you trying to simulate?  This failure was caused by two onConfigure calls coming in.  That can be simulated with the existing test framworks although it is only meaningful in the context of a particular plugin which many or may not handle it correctly.

Reply:
Done. https://github.com/istio/istio/issues/18396